src/utils/helpers/syncHelper.ts

```diff
@@ -0,0 +1,202 @@
+                        p1Id: el.intervention_id,
+                        p2Id: trees.tree_id,
+                    })
+                }
+            });
+        }
+    });
+    return quee
+}
+
+export const getPostBody = async (r: QueeBody) => {
+    if (r.type === 'intervention') {
+        const InterventionD = appRealm.objectForPrimaryKey<InterventionData>(RealmSchema.Intervention, r.p1Id);
+        return convertInterventiontoBody(JSON.parse(JSON.stringify(InterventionD)))
+    }
+    if (r.type === 'singleTree') {
+        const SingleTree = appRealm.objectForPrimaryKey<InterventionData>(RealmSchema.Intervention, r.p1Id);
+        return convertTreetoBody(JSON.parse(JSON.stringify(SingleTree)), JSON.parse(JSON.stringify(SingleTree.sample_trees[0])))
+    }
+    if (r.type === 'sampleTree') {
+        const TreeDetails = appRealm.objectForPrimaryKey<SampleTree>(RealmSchema.TreeDetail, r.p2Id);
+        const Intervention = appRealm.objectForPrimaryKey<InterventionData>(RealmSchema.Intervention, TreeDetails.intervention_id);
+        if (Intervention.location_id === '') {
+            return null
+        }
+        return convertTreetoBody(Intervention, TreeDetails)
+    }
+    if (r.type === 'treeImage') {
+        const TreeDetails = appRealm.objectForPrimaryKey<SampleTree>(RealmSchema.TreeDetail, r.p2Id);
+        if (TreeDetails.sloc_id === '') {
+            return null
+        }
+        const base64Image = await getImageAsBase64(TreeDetails.image_url)
+        const body = {
+            imageFile: `data:image/png;base64,${base64Image}`,
+            locationId: TreeDetails.tree_type === 'sample' ? TreeDetails.sloc_id : TreeDetails.parent_id,
+            imageId: TreeDetails.image_data.coordinateID
+        };
+        return body
+    }
+}
+
+export const convertInterventiontoBody = (d: InterventionData) => {
+    const metaData = JSON.parse(d.meta_data);
+    const interventionForm = setUpIntervention(d.intervention_key)
+    const postData: any = {
+        type: d.intervention_type,
+        captureMode: "on-site",
+        deviceLocation: metaData.app.deviceLocation,
+        geometry: {
+            type: d.location_type,
+            coordinates: d.location_type === 'Point' ? JSON.parse(d.location.coordinates) : [JSON.parse(d.location.coordinates)]
+        },
+        registrationDate: postTimeConvertor(d.intervention_date),
+        metadata: {
+            public: [metaData.public],
+            private: [metaData.private],
+            app: metaData.app
+        },
+    }
+    if (d.project_id) {
+        postData.plantProject = d.project_id
+    }
+    if (d.site_id) {
+        postData.plantProjectSite = d.site_id
+    }
+    if (interventionForm.species_required) {
+        const planted_species = d.planted_species.map(el => {
+            const species: any = {}
+            species.treeCount = el.count
+            if (el.guid === 'undefined') {
+                species.otherSpecies = "Unkown"
```

Comment:
I wonder if this spelling error of `Unkown` might result in a malfunction of the app?